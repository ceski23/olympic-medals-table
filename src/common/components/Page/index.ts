export { Page } from "./Page";
export type { CountryMedalsData } from "./Page";
