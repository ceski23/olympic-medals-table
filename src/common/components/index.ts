export * from "./Select";
export * from "./Input";
export * from "./Page";
export * from "./App";
