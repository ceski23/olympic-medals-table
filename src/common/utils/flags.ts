export const flags = Object.fromEntries(
  Object.entries(
    import.meta.glob("../../../node_modules/country-flag-icons/3x2/*.svg", {
      as: "url",
      eager: true,
    })
  ).map(([id, url]) => [id.match(/([A-Z][A-Z])\.svg/)?.[1] ?? id, url])
);
