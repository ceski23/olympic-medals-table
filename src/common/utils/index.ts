export * from "./guards";
export * from "./styles";
export * from "./flags";
