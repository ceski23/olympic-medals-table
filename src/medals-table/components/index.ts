export * from "./CountryFlag";
export * from "./MedalsTable";
export * from "./SortingIndicator";
export * from "./Table";
export * from "./CountryForm";
