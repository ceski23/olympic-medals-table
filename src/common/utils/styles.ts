import { Theme } from "@emotion/react";
import { Breakpoint } from "common/styles";

export const mediaQueryUp =
  (breakpoint: Breakpoint) =>
  ({ theme }: { theme: Theme }) =>
    `@media (min-width: ${theme.breakpoints[breakpoint] + 0.01}px)`;

export const mediaQueryUpTo =
  (breakpoint: Breakpoint) =>
  ({ theme }: { theme: Theme }) =>
    `@media (max-width: ${theme.breakpoints[breakpoint]}px)`;
