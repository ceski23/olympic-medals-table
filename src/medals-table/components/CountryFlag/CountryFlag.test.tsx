import { CountryFlag } from "medals-table/components";
import { renderWithProviders } from "test/utils";

describe("CountryFlag", () => {
  it("should render correctly", () => {
    const { asFragment } = renderWithProviders(<CountryFlag code="PL" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
