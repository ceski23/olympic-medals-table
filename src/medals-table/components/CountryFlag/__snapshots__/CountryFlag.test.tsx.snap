// Vitest Snapshot v1

exports[`CountryFlag > should render correctly 1`] = `
<DocumentFragment>
  <img
    alt="PL"
    height="20px"
    loading="lazy"
    src="/node_modules/country-flag-icons/3x2/PL.svg"
    width="30px"
  />
</DocumentFragment>
`;
